GCEWindowsAgent/addresses.go

```diff
@@ -372,3 +379,64 @@ func (a *addressMgr) set() error {
 
 	return nil
 }
+
+var osrelease release
+
+func enableNetworkInterfaces(ifaces []net.Interface) error {
+	var interfaces []string
+	for _, ni := range newMetadata.Instance.NetworkInterfaces {
+		iface, err := getInterfaceByMAC(ni.Mac, ifaces)
+		if err != nil {
+			if !containsString(ni.Mac, badMAC) {
+				logger.Errorf("Error getting interface: %s", err)
+				badMAC = append(badMAC, ni.Mac)
+			}
+			continue
+		}
+		interfaces = append(interfaces, iface.Name)
+	}
+
+	if osrelease.os == "rhel" && *osrelease.version.major == 7 {
```

Comment:
Would a switch with the os types broken out into separate functions make more sense? I think it would make it more readable, especially with the fact that suse returns early but rhel doesn't, and then a default setting just to make it clear all other types don't have anything additional.